include/RAJA/forallN_generic.hxx

```diff
@@ -46,153 +46,65 @@
 #ifndef RAJA_forallN_generic_HXX__
 #define RAJA_forallN_generic_HXX__
 
+#include <RAJA/LegacyCompatibility.hxx>
 #include "forallN_generic_lf.hxx"
```

Comment:
The reason there are XXX.hxx and XXX_lf.hxx files was because the XXX.hxx was code-gen'ed, and the XXX_lf.hxx wasn't (at some point lf stoof for "lambda-free", but that doesn't really matter).

Do you want to merge forallN_generic_lf.hxx into this file?
Same for forallN_permute_lf.hxx